packages/ember/src/components/form/select.gts

```diff
@@ -0,0 +1,412 @@
+    }
+    const childElements = Array.from(
+      document.querySelectorAll(`#${CSS.escape(this.menuId)} > li`),
+    );
+    const activeElement = childElements[this.activeItem];
+    if (!activeElement) {
+      return;
+    }
+    activeElement.scrollIntoView({ block: 'nearest' });
+  }
+
+  @action
+  moveUp(evt: KeyboardEvent) {
+    evt.preventDefault();
+    evt.stopPropagation();
+    this.internalSearchBuffer = '';
+    if (this.activeItem > 0) {
+      this.activeItem--;
+      this.scrollActiveItemIntoView();
+    }
+  }
+
+  @action
+  moveDown(evt: KeyboardEvent) {
+    evt.preventDefault();
+    evt.stopPropagation();
+    this.internalSearchBuffer = '';
+    if (this.activeItem < this.internalOptions.length - 1) {
+      this.activeItem++;
+      this.scrollActiveItemIntoView();
+    }
+  }
+
+  @action
+  enterKeyHandler(evt?: KeyboardEvent) {
+    evt?.preventDefault();
+    evt?.stopPropagation();
+    this.internalSearchBuffer = '';
+    if (!this.isOpen) {
+      this.onFocus();
+    } else {
+      this.exitKeyHandler();
+    }
+  }
+
+  @action
+  exitKeyHandler(evt?: KeyboardEvent) {
+    const optionsLength = this.internalOptions.length;
+    const validRange = this.activeItem >= 0 && this.activeItem < optionsLength;
+    evt?.preventDefault();
+    evt?.stopPropagation();
+    this.internalSearchBuffer = '';
+    if (validRange) {
+      const option = this.internalOptions[this.activeItem];
+      if (option != undefined) {
+        this.onSelectInternal(option);
+        return;
+      }
+    }
+    this.onBlur();
+  }
+
+  <template>
+    <button
+      class={{this.classList}}
+      type="button"
+      role="combobox"
+      disabled={{this.disabled}}
+      aria-controls={{this.menuId}}
+      aria-expanded={{this.isOpen}}
+      aria-haspopup="listbox"
+      {{on "click" this.toggleSelect}}
+      {{on "blur" this.onBlur}}
+      {{on "keydown" this.onKeyboardInput}}
+      {{onKey "ArrowUp" this.moveUp onlyWhenFocused=true}}
+      {{onKey "ArrowDown" this.moveDown onlyWhenFocused=true}}
+      {{onKey "Enter" this.enterKeyHandler onlyWhenFocused=true}}
+      {{onKey "Space" this.enterKeyHandler onlyWhenFocused=true}}
+      {{onKey "NumpadEnter" this.enterKeyHandler onlyWhenFocused=true}}
+      {{onKey "Tab" this.exitKeyHandler onlyWhenFocused=true}}
+      {{onKey "Escape" this.exitKeyHandler onlyWhenFocused=true}}
+      ...attributes
+    >
+      <span class="selected-display">
+        {{#if this.hasSelected}}
+          {{#if (has-block "display")}}
+            {{yield this.selected.raw to="display"}}
+          {{else if (has-block "option")}}
+            {{yield this.selected.raw to="option"}}
+          {{else}}
+            {{this.selected.label}}
+          {{/if}}
+        {{else}}
+          {{#if (has-block "empty")}}
+            {{yield to="empty"}}
+          {{else}}
+            {{this.defaultText}}
+          {{/if}}
+        {{/if}}
+      </span>
+
+      {{#if @loading}}
```

Comment:
Do you foresee any issues with having the select be open (and the options clickable) while `loading` is true?

Reply:
I think I've addressed this issue by making `isOpen = _isOpen & !loading`.
If the select is already open, it should close when it starts loading.